Update & add translations for Supabase README

lyqht opened the pull request:
## What kind of change does this PR introduce?

**Docs update: add + update translations for Supabase README.** 

These translations are generated using [DeepL](https://www.deepl.com/why-deepl-pro/), an AI translation service that is pretty accurate. You can refer to [this project](https://github.com/lyqht/deepL-demo) for how the translations for the Supabase README get generated. 

The TOCs at `languages.md` & `README.md` that points to the translations are manually updated at the moment.

## Additional context

Quite a few existing translation files were outdated, so the PR includes updates to them.

If the supabase maintainers prefer that the PR only adds translations and not update the existing translations, feel free to remove the changes to the updated files.

<details> <summary> Anything else? </summary>

I'll be working on a GitHub action to dynamically update these translation files whenever the main README.md gets updated. I can create another PR for that as well on Supabase if the maintainers like the idea of this. :D

</details>

lyqht committed:
add translations for Supabase README

lyqht committed:
remove english variants of readme

lyqht committed:
add new languages to i18n navigation toc

lyqht committed:
update links for translated readme

gregnr commented:
Thanks @lyqht for taking the time to do this 🚀 and really interesting idea to automate this via GitHub Action. Currently discussing this with some team members - will keep you posted!

gregnr commented on i18n/README.pt-br.md:
```diff
@@ -1,82 +1,83 @@
-- [x] Banco de dados hospedado de Postgres. [Docs](https://supabase.com/docs/guides/database)
-- [x] Autenticação e Autorização. [Docs](https://supabase.com/docs/guides/auth)
-- [x] APIs auto-geradas.
+- [Banco de dados Postgres hospedado. [Docs](https://supabase.com/docs/guides/database)
+- [x] Autenticação e autorização. [Docs](https://supabase.com/docs/guides/auth)
+- [x] APIs geradas automaticamente.
```
Hey @lyqht - something might have gone wrong with the first check box here (partially removed). Any ideas?

gregnr commented on i18n/README.pt-br.md:
```diff
@@ -182,7 +174,7 @@ Nossa biblioteca de cliente é modular. Cada sub-biblioteca é uma implementaç
   </tr>
   <tr>
-    <td>Rust</td>
+    <td>Ferrugem</td>
```
@lyqht Ideally this would have remained as `Rust` in all languages. Does DeepL support any kind of word tagging to prevent this?

gregnr commented:
Hey @lyqht really like this idea overall. Just a couple points I've noticed and would love your thoughts on - once addressed we can merge this.

I would love to see a PR with the GitHub Action if you are up for it. Happy to help collaborate with you on that and get keys, etc set up.

lyqht commented:
Thanks @gregnr for the good catch and helping to check the translated file! 😊

For the 1st issue on the checkbox, it seems to only happen for specific languages: `nl, pl, pt-BR, pt-PT`. The rest looks like they don't experience this issue. If it is ok with you, I'll manually update them for now and raise this issue with the DeepL help center since it seems to be a edge case problem with their translation.

For the 2nd issue, it's quite funny that the programming language is translated! I'll work on a way so that specific terms are not translated.

lyqht committed:
update readmes

lyqht commented:
hi @gregnr, 

- I've done the manual fix for the first issue on checkbox formatting. 
- There were also some table formatting issues on some other languages that I've fixed by changing the tag handling from `html` to `xml` on the deepL api. There's 1 consequence in doing so but it's pretty minor: markdown comment blocks like `<!-- comment -->` -> won't be translated.
- For the 2nd issue on excluding specific terms, there's a few ways to make sure specific elements are not translated. I've tested a few of them, and a hack that worked is to replace terms not to be ignored with `<keep> {term} </keep>` so that the translation engine doesn't translate it when using XML handling. 
 
The READMEs have been updated.

gregnr commented on i18n/README.pt-br.md:
```diff
@@ -152,19 +153,10 @@ Nossa biblioteca de cliente é modular. Cada sub-biblioteca é uma implementaç
     <td><a href="https://github.com/supabase-community/supabase-kt/tree/master/Realtime" target="_blank" rel="noopener noreferrer">realtime-kt</a></td>
     <td><a href="https://github.com/supabase-community/supabase-kt/tree/master/Storage" target="_blank" rel="noopener noreferrer">storage-kt</a></td>
-    <td><a href="https://github.com/supabase-community/supabase-kt/tree/master/Functions" target="_blank" rel="noopener noreferrer">functions-kt</a></td>
+    <td><a href="https://github.com/supabase-community/supabase-kt/tree/master/Functions" target="_blank" rel="noopener noreferrer">funções-kt</a></td>
```
These should also be excluded from renames

gregnr commented on i18n/README.pt-br.md:
```diff
@@ -100,16 +101,16 @@ Nossa biblioteca de cliente é modular. Cada sub-biblioteca é uma implementaç
   <th colspan="7">⚡️ Oficial ⚡️</th>
   <tr>
-    <td>JavaScript (TypeScript)</td>
+    <td><keep>Java</keep>Script (TypeScript)</td>
```
Would it make sense to wrap the entire text?
```suggestion
    <td><keep>JavaScript (TypeScript)</keep></td>
```

gregnr reviewed (commented):
Thanks @lyqht and nice idea with `<keep>` 👍 Would we need to manually add this to each new translation, or just add to the main `README.md`? I see the main `README.md` doesn't currently include the `<keep>`s.

lyqht committed:
update readmes to remove <keep> elements, and don't translate table that contains the programming languages

lyqht commented:
Thanks for helping to check again @gregnr 😊 I've done 2 things to fix the issue:
- Update the original README to have `<!-- notranslate --> and `<!-- /notranslate --> comments. I've added them to enclose the table rows that contains the programming languages that should not be translated.
- I've updated script in the [demo project](https://github.com/lyqht/deepL-demo) so that the text within that enclosed area won't be translated, and the ugly `<keep></keep>` won't be left in the translated documents.

gregnr commented on i18n/README.pt-br.md:
```diff
@@ -1,91 +1,93 @@
-- [x] Banco de dados hospedado de Postgres. [Docs](https://supabase.com/docs/guides/database)
-- [x] Autenticação e Autorização. [Docs](https://supabase.com/docs/guides/auth)
-- [x] APIs auto-geradas.
+- [x] [Banco de dados Postgres hospedado. [Docs](https://supabase.com/docs/guides/database)
```
We have one more phantom bracket here.

gregnr reviewed (commented):
This is looking really good, thanks for the hard work @lyqht. Looks like we just have one more phantom bracket in there. I think that's all that's left before we can merge this!

lyqht committed:
remove bracket

gregnr commented on i18n/README.it.md:
```diff
@@ -1,85 +1,260 @@
-- [x] Sottoscrizioni in tempo reale
-- [x] Autenticazione e autorizzazioni
-- [x] API generate automaticamente
+- [Database Postgres ospitato. [Documenti](https://supabase.com/docs/guides/database)
```
```suggestion
- Database Postgres ospitato. [Documenti](https://supabase.com/docs/guides/database)
```

gregnr commented on i18n/README.es.md:
```diff
@@ -1,82 +1,85 @@
-- [x] Almacenamiento
+[Supabase](https://supabase.com) es una alternativa de código abierto a Firebase. Estamos construyendo las características de Firebase utilizando herramientas de código abierto de nivel empresarial.
+
+- [Base de datos Postgres alojada. [Documentación](https://supabase.com/docs/guides/database)
```
```suggestion
- Base de datos Postgres alojada. [Documentación](https://supabase.com/docs/guides/database)
```

gregnr commented on i18n/README.cs.md:
```diff
@@ -0,0 +1,260 @@
+
+[Supabase](https://supabase.com) je open source alternativa Firebase. Vytváříme funkce Firebase pomocí open source nástrojů podnikové třídy.
+
+- hostovaná databáze Postgres [x]. [Dokumenty](https://supabase.com/docs/guides/database)
```
```suggestion
- [x] hostovaná databáze Postgres. [Dokumenty](https://supabase.com/docs/guides/database)
```

gregnr commented on i18n/README.nl.md:
```diff
@@ -1,87 +1,260 @@
+- [x]Gehoste Postgres Database. [Docs](https://supabase.com/docs/guides/database)
+- [x] Authenticatie en Autorisatie. [Docs](https://supabase.com/docs/guides/auth)
+- [x] Automatisch gegenereerde API's.
+  - REST. [Docs](https://supabase.com/docs/guides/database/api#rest-api)
+  - GraphQL. [Docs](https://supabase.com/docs/guides/database/api#graphql-api)
```
```suggestion
  - [x] REST. [Docs](https://supabase.com/docs/guides/database/api#rest-api)
  - [x] GraphQL. [Docs](https://supabase.com/docs/guides/database/api#graphql-api)
```

gregnr commented on i18n/README.sk.md:
```diff
@@ -0,0 +1,260 @@
+
+[Supabase](https://supabase.com) je open source alternatíva Firebase. Funkcie Firebase budujeme pomocou open source nástrojov podnikovej triedy.
+
+- hostovaná databáza Postgres [x]. [Dokumenty](https://supabase.com/docs/guides/database)
```
```suggestion
- [x] hostovaná databáza Postgres. [Dokumenty](https://supabase.com/docs/guides/database)
```

gregnr committed:
fix(readme): translation formatting errors

lyqht commented:
Thanks @greg for making fixes!

Took a look at the failing checks:

- We should configure reviewdog not to run on i18n files that are not English
- im not too sure if prettier should run on these files

gregnr committed:
Merge branch 'master' into lyqht/i18n

gregnr committed:
chore: ignore i18n for spell check

lyqht committed:
modify avoid-typos github action to exclude only READMEs in i18n folder

lyqht committed:
prettier formatting

gregnr committed:
fix: spell check exclude directories
